### ISSUE TITLE
[URGENT!] App under iOS <14.0 crashes on Voice.start()

### ISSUE THREAD

**Issue by MedRedha** [react-native-voice/voice#289]
Greetings beautiful people,

I'm having a big issue if someone can help me please 🙏  the package works perfectly under iOS >14.0 but immediately crashes when I try to build on an inferior one (i.e. 13.6). I tried my best to debug it and here's what I'm getting from xCode console : 

`2021-01-27 14:45:44.620057+0100 LegalDoctrineAlpha[75986:1310434] [aqme] AQME.h:254:IOProcFailure: AQDefaultDevice (1): output stream 0: null buffer
2021-01-27 14:45:44.620305+0100 LegalDoctrineAlpha[75986:1310434] [aqme] AQMEIO_HAL.cpp:1774:IOProc: EXCEPTION thrown (-50): error != 0`

and here is a screenshot of the line pointing to it.
![image_2021-01-27_145143](https://user-images.githubusercontent.com/49821074/106000503-2c144000-60af-11eb-9a6e-ca682511671d.png)

![image_2021-01-27_145309](https://user-images.githubusercontent.com/49821074/106000660-5ebe3880-60af-11eb-83ca-67d762c675c5.png)

**Comment by YesSkyscrapers** [comment]
yo, face the same issue...
Did u find any solution?

**Comment by MedRedha** [comment]
> yo, face the same issue...
> Did u find any solution?

Not yet bro 😞 I disabled the mic button for iOS <14.0 waiting for a fix or something

**Comment by YesSkyscrapers** [comment]
Hey man!
The problem was unexpected for me, cuz it worked perfectly just a week ago...
I almost believed it happened cuz of MacOS's update to Big Sur, but today a little trick was able to fix everything...
I was compiling the release via the command line, when the iPhone XR simulator was selected in xcode... 

![Снимок экрана 2021-02-02 в 10 54 24](https://user-images.githubusercontent.com/45424671/106568956-07074d80-6545-11eb-986c-cb15f1c081a7.png)

I looked at the simulator's ios version...  there was ios 12.0...

![Снимок экрана 2021-02-02 в 10 56 32](https://user-images.githubusercontent.com/45424671/106569223-52216080-6545-11eb-97ed-688e4720eace.png)

Then I added the same iPhone but with the latest version of iOS 14.4. 
I selected new simulator and built it to check that the bug doesn't reproduce (cuz ios >=14)... 

![Снимок экрана 2021-02-02 в 10 57 38](https://user-images.githubusercontent.com/45424671/106569369-7a10c400-6545-11eb-9e7b-cc3df2ee8585.png)

After that, I repeated build through the command line.... The bug wasn't reproduced on any of the phones where it was reproduced recently, namely:
5S iOS 12.0
XR iOS 14.3
SE1 iOS 13.7

Idk how it relates, but the bug isn't reproduced now...

**Comment by inveon-farukhasanefendioglu** [comment]
I tried the same steps with @YesSkyscrapers but no luck at all...

Still having a crash same as @MedRedha mentioned under iOS 14.x

Any different workaround or fixes would be perfect. I could not make that function work under iOS 14.x both on emulator and real device.

**Comment by safaiyeh** [comment]
Hm is there any documentation referencing a change to these libraries for iOS 14?

**Comment by MedRedha** [comment]
> Hm is there any documentation referencing a change to these libraries for iOS 14?

Precisely! I checked everywhere and nothing!! There is no reason for the package or for the usage of the speech recognition to crash or bug and that's what is weird 😞

**Comment by safaiyeh** [comment]
@MedRedha Could you create a small demo app with the latest version of voice and RN so I could check it out and debug it.

**Comment by josemojena** [comment]
I have the same problem but only if connect the ios earphones, has anyone find a solution?

**Comment by safaiyeh** [comment]
> I have the same problem but only if connect the ios earphones, has anyone found a solution?

I've seen this reported a few times where earphones messes things up. https://github.com/react-native-voice/voice/pull/294 is one example. Could you make a separate issue for this?

**Comment by forresthopkinsa** [comment]
Could be related to [this problem](https://www.reddit.com/r/iOSProgramming/comments/jff05e/xcode_121_problem_with_audio_using_simulator_with/)

**Comment by rodolfobarretoweb** [comment]
Hey guys.

I was having the same issue reported here and the only thing that fix this problem was putting the code below in the info.plist file as documented in the installation guide.

<key>NSMicrophoneUsageDescription</key>
<string>Description of why you require the use of the microphone</string>
<key>NSSpeechRecognitionUsageDescription</key>
<string>Description of why you require the use of the speech recognition</string>

**Comment by MedRedha** [comment]
> Hey guys.
> 
> 
> 
> I was having the same issue reported here and the only thing that fix this problem was putting the code below in the info.plist file as documented in the installation guide.
> 
> 
> 
> ```
> 
> <key>NSMicrophoneUsageDescription</key>
> 
> <string>Description of why you require the use of the microphone</string>
> 
> <key>NSSpeechRecognitionUsageDescription</key>
> 
> <string>Description of why you require the use of the speech recognition</string>
> 
> ```

Thanks a lot buddy! I'm gonna try it and check if it fixes the issue for me as well 🙏

**Comment by rodolfobarretoweb** [comment]
> > Hey guys.
> > I was having the same issue reported here and the only thing that fix this problem was putting the code below in the info.plist file as documented in the installation guide.
> > ```
> > 
> > <key>NSMicrophoneUsageDescription</key>
> > 
> > <string>Description of why you require the use of the microphone</string>
> > 
> > <key>NSSpeechRecognitionUsageDescription</key>
> > 
> > <string>Description of why you require the use of the speech recognition</string>
> > ```
> 
> Thanks a lot buddy! I'm gonna try it and check if it fixes the issue for me as well 🙏

Let me know if works for you too.

**Comment by anujmpec** [comment]
@DeepakM02  link is not working.
I am also facing this issue. Any solution?

**Comment by DeepakM02** [comment]
Hey @anujmpec You can check this link now.

**Comment by Anujmoglix** [comment]
@DeepakM02  please share the link.

**Comment by DeepakM02** [comment]
@Anujmoglix you can use this 
https://github.com/react-native-voice/voice/issues/289#issuecomment-946662633

**Comment by samuelzxu** [comment]
I'm still facing this issue despite using version 3.2.4

**Comment by safaiyeh** [comment]
This project has been deprecated and archived. Please migrate to [`expo-speech-recognition`](https://github.com/jamsch/expo-speech-recognition).
Thank you for your contributions! ❤️
